### ISSUE TITLE
Retry support for for asynchronous requests.

### ISSUE THREAD

**Issue by david-romero** [polysantiago/spring-boot-rest-client#2]
As spring-boot-rest-client supports spring-retry and asynchronous requests we can add retry support for for asynchronous requests.

What do you think? Can I send a PR with this feature?

**Comment by david-romero** [comment]
What do you think @polysantiago ??

**Comment by polysantiago** [comment]
Yeah! go ahead @david-romero! I can have a look as soon as time permits :)
